Some new recipes and updates

khallebal opened the pull request:
The reason i switched to cmake build system in the libsndfile recipe is to provide us with the sndfileconfig.cmake file which is needed by the libkdegames pkg, and the reason i used the commit hash in this update is because the cmake build system was broken and had only been fixed recently in mid december, i'll switch back to tarballs again by the next offcial release.
Thanks for reviewing.

alaviss commented on games-puzzle/kpat/kpat-17.12.0.recipe:
```diff
@@ -0,0 +1,140 @@
+		-DECM_DIR=/system/data/cmake/Modules/ECM/cmake \
+		-DCMAKE_INSTALL_PREFIX=$appsDir/kpat \
+		-DKDE_INSTALL_BINDIR=$appsDir/kpat \
+		-DKDE_INSTALL_LIBDIR=$appsDir/kpat/lib \
```
Please use `$libDir` instead

```diff
-		-DKDE_INSTALL_LIBDIR=$appsDir/kpat/lib \
+		-DKDE_INSTALL_LIBDIR=$libDir \

diversys replied:
If this library isn't used by other apps I think it's better to live in kpat folder.

khallebal replied:
The app can't find it anyway if it's installed in "$libDir", it doesn't work.

fbrosson replied:
I did not test it, but I wonder if it would work with these changes:
```diff
- 		-DCMAKE_INSTALL_PREFIX=$appsDir/kpat \
+ 		-DCMAKE_INSTALL_PREFIX=$prefix \
-		-DKDE_INSTALL_BINDIR=$appsDir/kpat \
+		-DCMAKE_INSTALL_BINDIR=$appsDir \
-		-DKDE_INSTALL_LIBDIR=$appsDir/kpat/lib \
+		-DCMAKE_INSTALL_LIBDIR=$libDir \
```
(You would also need to modify `INSTALL()`.)

khallebal replied:
It doesn't really matter whether you use -DCMAKE or -DKDE i tried both, either way the app seems only to look for it in $appsDir/KPatience/lib.

alaviss commented on games-puzzle/kpat/kpat-17.12.0.recipe:
```diff
@@ -0,0 +1,140 @@
+To win the game a player has to arrange a single deck of cards \
+in certain order amongst each other. "
+HOMEPAGE="https://edu.kde.org/kpat"
+COPYRIGHT="2010-2017 KDE Organisation"
```
There's an [another copyright holder][0]

```diff
-COPYRIGHT="2010-2017 KDE Organisation"
+COPYRIGHT="2010-2017 KDE Organisation
+	1995 Paul Olav Tvete"
```

[0]: https://github.com/KDE/kpat/blob/v17.12.1/COPYING#L17

alaviss commented on games-puzzle/kpat/kpat-17.12.0.recipe:
```diff
@@ -0,0 +1,140 @@
+in certain order amongst each other. "
+HOMEPAGE="https://edu.kde.org/kpat"
+COPYRIGHT="2010-2017 KDE Organisation"
+LICENSE="GNU LGPL v2"
```
There seems to be [a custom license][0] around as well. Please add it.

If this package also contains the documentations, they're subjected to [GNU FDL][1].

[0]: https://github.com/KDE/kpat/blob/v17.12.1/COPYING#L17
[1]: https://github.com/KDE/kpat/blob/v17.12.1/COPYING.DOC#L1

alaviss commented:
For media-libs/libsndfile, create a new recipe with version `1.0.28~git` as you're not using the official release.

diversys commented on games-puzzle/kpat/kpat-17.12.0.recipe:
```diff
@@ -0,0 +1,140 @@
+
+	cmake .. \
+		-DECM_DIR=/system/data/cmake/Modules/ECM/cmake \
+		-DCMAKE_INSTALL_PREFIX=$appsDir/kpat \
```
-DCMAKE_INSTALL_PREFIX=$appsDir/KPatience

alaviss commented on games-puzzle/kpat/kpat-17.12.1.recipe:
```diff
@@ -0,0 +1,145 @@
+{
+	# disable docs generation until libxml package is fixed
+	sed -i s/DocTools/#DocTools/g CMakeLists.txt
+	sed -i s/add_subdirectory\(doc\)/#add_subdirectory\(doc\)/g CMakeLists.txt
```
Please use quotes instead. Also, this style is used more often in HaikuPorts.

Please consider moving this to the `PATCH()` function.

```diff
-	sed -i s/DocTools/#DocTools/g CMakeLists.txt
-	sed -i s/add_subdirectory\(doc\)/#add_subdirectory\(doc\)/g CMakeLists.txt
+	sed -e 's/DocTools/#DocTools/g' \
+		-e 's/add_subdirectory(doc)/#add_subdirectory(doc)/g' \
+		-i CMakeLists.txt

khallebal replied:
Well, i've seen so many recipes doing so, i just followed the same pattern.

alaviss replied:
@fbrosson Can you give us some metrics?

fbrosson replied:
(Sorry for this late reply.)
I think most of our recipes use double quotes. We probably also have some with single quotes.
I like to use double quotes because it's more eye-friendly.
I can/will do some metrics, but a bit later. (There are lots of PRs waiting to be reviewed or merged...)

alaviss commented on games-puzzle/kpat/kpat-17.12.1.recipe:
```diff
@@ -0,0 +1,145 @@
+	cmd:gettext$secondaryArchSuffix
+	cmd:make
+	cmd:python2
+	cmd:update_mime_database$secondaryArchSuffix
```
This might save you some variables.

From [KDEInstallDirs][0]:

> If `KDE_INSTALL_USE_QT_SYS_PATHS` is set to TRUE before including this module, the default values for some variables are instead queried from Qt5’s qmake (where mentioned in the parentheses above). If not set, **it will default to TRUE if Qt5’s qmake is found** and it’s `QT_INSTALL_PREFIX` is the same as `CMAKE_INSTALL_PREFIX`, otherwise default to FALSE. This variable should NOT be set from within CMakeLists.txt files, instead is intended to be set manually when configuring a project which uses KDEInstallDirs (e.g. by packagers).

```diff
+	cmd:qmake >= 5
```

[0]: https://api.kde.org/ecm/kde-module/KDEInstallDirs.html

khallebal replied:
I'm not sure i understand what you mean by "This might save you some variables."
From KDEInstallDirs:?
for the "cmd:qmake >= 5" i 've never been able to build anything with it when i add >=5 to it, it always says command not found.

alaviss replied:
According to `KDEInstallDirs` documentation (the cmake module that's used by this package). If you have qmake in your `PATH`, it could use qmake to get default values for some variable (namely LIBDIR, PREFIX,...)

Those usually have the package name in there so maybe this suggestion wasn't so good after all. So feel free to leave it out

fbrosson replied:
@khallebal, there are around 20 recipes which have `cmd:qmake$secondaryArchSuffix >= 5`, and I have recently built one of them: [CuteMarkEd](https://github.com/haikuports/haikuports/tree/master/app-text/cutemarked)

BTW, if you are running Haiku x86_gcc2 then you really need the `$secondaryArchSuffix`, otherwise you will get a `command not found` error.

alaviss replied:
@fbrosson It's not necessary. As I wrote [earlier][0], qmake is only used to fetch system prefix. This port uses CMake instead

[0]: https://github.com/haikuports/haikuports/pull/2116#discussion_r161420514

alaviss commented on games-puzzle/kpat/kpat-17.12.1.recipe:
```diff
@@ -0,0 +1,145 @@
+		-DECM_DIR=/system/data/cmake/Modules/ECM/cmake \
+		-DCMAKE_INSTALL_PREFIX=$appsDir/KPatience \
+		-DKDE_INSTALL_BINDIR=$appsDir/KPatience \
+		-DKDE_INSTALL_LIBDIR=$appsDir/KPatience/lib \
```
Note that unless you install the libraries to `$libDir`, you can't list them in `PROVIDES`.

P/s: You would want to clean up the work directory with `haikuporter -c` before trying to rebuild.

```diff
-		-DCMAKE_INSTALL_PREFIX=$appsDir/KPatience \
+		-DCMAKE_INSTALL_PREFIX=$prefix \
 		-DKDE_INSTALL_BINDIR=$appsDir/KPatience \
-		-DKDE_INSTALL_LIBDIR=$appsDir/KPatience/lib \
+		-DKDE_INSTALL_LIBDIR=$libDir \

khallebal replied:
>Note that unless you install the libraries to $libDir, you can't list them in PROVIDES.

Alright, i'm gonna fix that.

khallebal commented on games-puzzle/kpat/kpat-17.12.1.recipe:
```diff
@@ -0,0 +1,145 @@
+COPYRIGHT="2010-2017 KDE Organisation
+			1995 Paul Olav Tvete"
+LICENSE="GNU LGPL v2
+			POT v2"
```
>why is there a qmplay2 patchset lying around?
I thought i'd removed it but it's still hangin' around, sometimes i find "git rm" and "git mv" don't do wht they are suppose to do.
For the license i couldn't find it in the source archive, it's just part of the GNU license, also i tried adding it as you mentioned above already it kept complaining about no matching license, so i think we should just skip it for now, until everything else is set.

alaviss replied:
You would want to add this text to a file named `POT v2` (mind the space) in the `licenses` folder next to your recipe

> Copyright (C) 1995 Paul Olav Tvete \<paul@troll.no\>
>
> Permission to use, copy, modify, and distribute this software and its
documentation for any purpose and without fee is hereby granted,
provided that the above copyright notice appear in all copies and that
both that copyright notice and this permission notice appear in
supporting documentation.
>
> This file is provided AS IS with no warranties of any kind.  The author
shall have no liability with respect to the infringement of copyrights,
trade secrets or any patents by this file or any part thereof.  In no
event will the author be liable for any lost revenue or profits or
other special, indirect and consequential damages.

khallebal commented on games-puzzle/kpat/kpat-17.12.1.recipe:
```diff
@@ -0,0 +1,145 @@
+REVISION="1"
+SOURCE_URI="https://github.com/KDE/kpat/archive/v$portVersion.tar.gz"
+CHECKSUM_SHA256="5241e8092bf4820b4ad2a016fed0928ed12815edb050dbb3f5512103aca0f293"
+#PATCHES="kpat-$portVersion.patchset"
```
>Is the patch no longer necessary? If so, please remove this line and the patch from this PR.

No sorry about that, i just forgot to put it back when iwas trying out the version.

khallebal commented on games-puzzle/kpat/kpat-17.12.1.recipe:
```diff
@@ -0,0 +1,145 @@
+PROVIDES="
+	kpat$secondaryArchSuffix = $portVersion
+	app:kpat$secondaryArchSuffix
+	lib:libkcardgame$secondaryArchSuffix
```
Ok , will do

khallebal commented on kde-frameworks/libkdegames/libkdegames-17.12.1.recipe:
```diff
@@ -0,0 +1,144 @@
+PROVIDES="
+	libkdegames$secondaryArchSuffix = $portVersion
+	lib:libKF5KDEGames$secondaryArchSuffix = $portVersion
+	lib:libKF5KDEGamesPrivate$secondaryArchSuffix = $portVersion
```
Again, i just followed other recipes as templates.
I'll fix all this stuff tomorrow.

alaviss commented on app-crypt/gpgme/gpgme-1.9.0.recipe:
```diff
@@ -51,13 +54,14 @@ BUILD_REQUIRES="
 	haiku${secondaryArchSuffix}_devel
 	cmd:autoconf
 	cmd:automake
 	cmd:bison
```
Missing [`$secondaryArchSuffix`][0]

```diff
- 	cmd:bison
+ 	cmd:bison$secondaryArchSuffix
```

[0]: https://github.com/haikuports/haikuports/blob/cca35279c8f9111b346d153426894bfccf1b3e9a/sys-devel/bison/bison-3.0.4.recipe#L26

alaviss commented on app-crypt/gpgme/gpgme-1.9.0.recipe:
```diff
@@ -51,13 +54,14 @@ BUILD_REQUIRES="
+	devel:libQt5Core${secondaryArchSuffix}
 	"
 BUILD_PREREQUIRES="
 	haiku${secondaryArchSuffix}_devel
```
This should be moved to `BUILD_REQUIRES`

```diff
 BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
-	devel:libgpg_error${secondaryArchSuffix}
-	devel:libglib_2.0$secondaryArchSuffix
-	devel:libassuan${secondaryArchSuffix}
-	devel:libQt5Core${secondaryArchSuffix}
+	devel:libgpg_error$secondaryArchSuffix
+	devel:libglib_2.0$secondaryArchSuffix
+	devel:libassuan$secondaryArchSuffix
+	devel:libQt5Core$secondaryArchSuffix
 	"
 BUILD_PREREQUIRES="
-	haiku${secondaryArchSuffix}_devel
```

alaviss commented on app-crypt/gpgme/gpgme-1.9.0.recipe:
```diff
@@ -29,19 +29,22 @@ PROVIDES="
 	cmd:gpgme_tool${secondaryArchSuffix} = $portVersion compat >= 1.5
 	lib:libgpgme${secondaryArchSuffix} = 11.18.0 compat >= 11
 	lib:libgpgmepp${secondaryArchSuffix} = 6.4.0 compat >= 6
+	lib:libqgpgme${secondaryArchSuffix} = 7.2.0 compat >= 7
```
These versions should be converted to the latest format (and as expected, someone forgot to bump this since 1.5-1.6). Also, remove braces when unneeded.

```diff
 ARCHITECTURES="!x86_gcc2 x86 x86 x86_64"
 SECONDARY_ARCHITECTURES="x86"
 
+libVersion="11.18.0"
+libCPlusVersion="6.4.0"
+libQtVersion="7.2.0"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+libCPlusVersionCompat="$libCPlusVersion compat >= ${libCPlusVersion%%.*}"
+libQtVersionCompat="$libQtVersion compat >= ${libQtVersion%%.*}"
+portVersionCompat="$portVersion compat >= ${portVersion%.*}"
+
 #TODO: fix gpgme-config hardcoded package paths
 PROVIDES="
- 	gpgme${secondaryArchSuffix} = $portVersion compat >= 1.5
-	cmd:gpgme_tool${secondaryArchSuffix} = $portVersion compat >= 1.5
-	lib:libgpgme${secondaryArchSuffix} = 11.18.0 compat >= 11
-	lib:libgpgmepp${secondaryArchSuffix} = 6.4.0 compat >= 6
-	lib:libqgpgme${secondaryArchSuffix} = 7.2.0 compat >= 7
+ 	gpgme$secondaryArchSuffix = $portVersionCompat
+	cmd:gpgme_tool$secondaryArchSuffix = $portVersionCompat
+	lib:libgpgme$secondaryArchSuffix = $libVersionCompat
+	lib:libgpgmepp$secondaryArchSuffix = $libVersionCPlusCompat
+	lib:libqgpgme$secondaryArchSuffix = $libVersionQtCompat

alaviss commented on app-crypt/gpgme/gpgme-1.9.0.recipe:
```diff
@@ -29,19 +29,22 @@ PROVIDES="
 	lib:libgpg_error${secondaryArchSuffix}
 	lib:libglib_2.0$secondaryArchSuffix
 	lib:libassuan${secondaryArchSuffix}
+	lib:libQt5Core${secondaryArchSuffix}
```
```diff
-	haiku${secondaryArchSuffix}
-	lib:libgpg_error${secondaryArchSuffix}
+	haiku$secondaryArchSuffix
+	lib:libgpg_error$secondaryArchSuffix
 	lib:libglib_2.0$secondaryArchSuffix
-	lib:libassuan${secondaryArchSuffix}
-	lib:libQt5Core${secondaryArchSuffix}
+	lib:libassuan$secondaryArchSuffix
+	lib:libQt5Core$secondaryArchSuffix

alaviss commented on app-crypt/gpgme/gpgme-1.9.0.recipe:
```diff
@@ -29,19 +29,22 @@ PROVIDES="
 	cmd:gpgme_config${secondaryArchSuffix} = $portVersion compat >= 1.5
 	devel:libgpgme${secondaryArchSuffix} = 11.18.0 compat >= 11
 	devel:libgpgmepp${secondaryArchSuffix} = 6.4.0 compat >= 6
+	devel:libqgpgme${secondaryArchSuffix} = 7.2.0 compat >= 7
```
```diff
-	gpgme${secondaryArchSuffix}_devel = $portVersion compat >= 1.5
-	cmd:gpgme_config${secondaryArchSuffix} = $portVersion compat >= 1.5
-	devel:libgpgme${secondaryArchSuffix} = 11.18.0 compat >= 11
-	devel:libgpgmepp${secondaryArchSuffix} = 6.4.0 compat >= 6
-	devel:libqgpgme${secondaryArchSuffix} = 7.2.0 compat >= 7
+	gpgme${secondaryArchSuffix}_devel = $portVersionCompat
+	cmd:gpgme_config$secondaryArchSuffix = $portVersionCompat
+	devel:libgpgme$secondaryArchSuffix = $libVersionCompat
+	devel:libgpgmepp$secondaryArchSuffix = $libCPlusVersionCompat
+	devel:libqgpgme$secondaryArchSuffix = $libQtVersionCompat

alaviss commented on app-crypt/gpgme/gpgme-1.9.0.recipe:
```diff
@@ -66,6 +70,7 @@ BUILD_PREREQUIRES="
 	cmd:make
 	cmd:makeinfo
 	cmd:msgfmt
+	cmd:pkg_config${secondaryArchSuffix}
```
```diff
-	cmd:pkg_config${secondaryArchSuffix}
+	cmd:pkg_config$secondaryArchSuffix

alaviss commented on app-crypt/gpgme/gpgme-1.9.0.recipe:
```diff
@@ -51,13 +54,14 @@ BUILD_REQUIRES="
 	cmd:automake
 	cmd:bison
-	cmd:gcc${secondaryArchSuffix}
+	cmd:g++${secondaryArchSuffix}
```
```diff
-	cmd:g++${secondaryArchSuffix}
+	cmd:g++$secondaryArchSuffix

khallebal commented:
I can't see from where i am what's wrong, but it still looks like it failed, anything i need to change?

fbrosson commented on app-crypt/gpgme/gpgme-1.9.0.recipe:
```diff
@@ -15,57 +15,68 @@ LICENSE="
+	cmd:gpgme_config$secondaryArchSuffix = $portVersionCompat
+	devel:libgpgme$secondaryArchSuffix = $libVersionCompat
+	devel:libgpgmepp$secondaryArchSuffix = $libCPlusVersionCompat
+	devel:libqgpgme$secondaryArchSuffix = $libQtVersionCompat	"
```
```diff
-	devel:libqgpgme$secondaryArchSuffix = $libQtVersionCompat	"
+	devel:libqgpgme$secondaryArchSuffix = $libQtVersionCompat
	"
```

fbrosson commented on app-crypt/gpgme/gpgme-1.9.0.recipe:
```diff
@@ -15,57 +15,68 @@ LICENSE="
+	cmd:gpgme_tool$secondaryArchSuffix = $portVersionCompat
+	lib:libgpgme$secondaryArchSuffix = $libVersionCompat
+	lib:libgpgmepp$secondaryArchSuffix = $libVersionCPlusCompat
+	lib:libqgpgme$secondaryArchSuffix = $libVersionQtCompat	"
```
```diff
-	lib:libqgpgme$secondaryArchSuffix = $libVersionQtCompat	"
+	lib:libqgpgme$secondaryArchSuffix = $libVersionQtCompat
+	"
```

fbrosson commented on app-crypt/gpgme/gpgme-1.9.0.recipe:
```diff
@@ -15,57 +15,68 @@ LICENSE="
 	cmd:gpg_error$secondaryArchSuffix
-	cmd:ld${secondaryArchSuffix}
+	cmd:ld$secondaryArchSuffix
 	cmd:libtoolize
```
```diff
-	cmd:libtoolize
+	cmd:libtoolize$secondaryArchSuffix
```

BTW, should probably drop git:
```diff
-	cmd:git
```

alaviss commented on games-puzzle/kpat/kpat-17.12.1.recipe:
```diff
@@ -0,0 +1,139 @@
+
+	cmake .. \
+		-DECM_DIR=/system/data/cmake/Modules/ECM/cmake \
+		-DCMAKE_INSTALL_PREFIX=$prefix
```
```diff
-		-DCMAKE_INSTALL_PREFIX=$prefix
+		-DCMAKE_INSTALL_PREFIX=$prefix \

khallebal replied:
This is one is not necessary as /boot/system is already the default in Cmake.

alaviss replied:
Sorry, maybe an unified diff would let you see why

```diff
 
 	cmake .. \
 		-DECM_DIR=/system/data/cmake/Modules/ECM/cmake \
-		-DCMAKE_INSTALL_PREFIX=$prefix
+		-DCMAKE_INSTALL_PREFIX=$prefix \
 		-DKDE_INSTALL_BINDIR=$appsDir/KPatience \
 		-DKDE_INSTALL_LIBDIR=$appsDir/KPatience/lib \
 		-DKDE_INSTALL_DATADIR=$dataDir \
```

alaviss commented on games-puzzle/kpat/kpat-17.12.1.recipe:
```diff
@@ -0,0 +1,139 @@
+
+TEST()
+{
+	make test
```
Did you forgot to `cd build`?

khallebal replied:
Does it fail here?, i actually took an existing recipe and modified it, so it was already like that, and it builds just fine as it is locally.

fbrosson replied:
`TEST()` is only used if you run `haikuporter --test kpat`.

(BTW, before running this command, you need to have built the package. And if you call `haikuporter -c kpat` then you need to build it again.)

Regading the `cd build`, if it is needed, then you might wish to combine it with the `make test`:
```diff
-	cd build
-	make test
+	make -C build test
```

alaviss commented on games-puzzle/kpat/patches/kpat-17.12.1.patchset:
```diff
@@ -0,0 +1,64 @@
+-    DBusAddons
+-    DocTools
++#	DBusAddons
++#	DocTools
```
You could just remove them. The commit could easily be reverted if these KF5 libs are fixed

```diff
-+#	DBusAddons
-+#	DocTools

alaviss commented on games-puzzle/kpat/patches/kpat-17.12.1.patchset:
```diff
@@ -0,0 +1,64 @@
+ add_subdirectory(sounds)
+ add_subdirectory(themes)
+-add_subdirectory(doc)
++#add_subdirectory(doc)
```
```diff
-+#add_subdirectory(doc)

alaviss commented on games-puzzle/kpat/patches/kpat-17.12.1.patchset:
```diff
@@ -0,0 +1,64 @@
+ target_link_libraries(kpat
+     KF5::Crash
+-    KF5::DBusAddons
++#	KF5::DBusAddons
```
```diff
-+#	KF5::DBusAddons

alaviss commented on games-puzzle/kpat/patches/kpat-17.12.1.patchset:
```diff
@@ -0,0 +1,64 @@
+ #include <QDebug>
+ #include <KLocalizedString>
+-#include <KDBusService>
++//#include <KDBusService>
```
```diff
-+//#include <KDBusService>

alaviss commented on games-puzzle/kpat/patches/kpat-17.12.1.patchset:
```diff
@@ -0,0 +1,64 @@
+     w->show();
+ 
+-    const KDBusService dbusService(KDBusService::Multiple);
++//    const KDBusService dbusService(KDBusService::Multiple);
```
```diff
-+//    const KDBusService dbusService(KDBusService::Multiple);

khallebal replied:
There is no need to remove them completely, we will just drop the patchset once the libs are fixed, what do you think?

alaviss replied:
Sure, but I prefer small diff, the changes will be more obvious. But feel free to keep it as you wish, this is more or less only a suggestion

alaviss commented on kde-frameworks/kdnssd/kdnssd-5.41.0.recipe:
```diff
@@ -0,0 +1,78 @@
+	cmd:cmake
+	cmd:g++$secondaryArchSuffix
+	cmd:make
+	#cmd:python2
```
Please remove if unneeded

```diff
-	#cmd:python2

khallebal replied:
python is a build dep for this package.

alaviss replied:
Then please remove the `#`

korli commented on app-crypt/gpgme/gpgme-1.9.0.recipe:
```diff
@@ -15,57 +15,69 @@ LICENSE="
 	lib:libglib_2.0$secondaryArchSuffix
-	lib:libassuan${secondaryArchSuffix}
+	lib:libassuan$secondaryArchSuffix
+	lib:libQt5Core$secondaryArchSuffix
```
Please use a subpackage for Qt5 stuff.

korli commented:
Could you please split this PR? so many changes it's not manageable anymore.

khallebal commented:
@alaviss , @korli , i had already removed the braces in my last commit for the 

> lib:libassuan${secondaryArchSuffix}

and the same for 

> #cmd:python2

I don't have those chars in my recipe, hopefully git will add the changes correctly this time?
Korli for your suggestions, how do i split the PR? and how do i make a subpackage for libQgpgme? i'm still not very familliar with git/github.

fbrosson commented:
@khallebal, there is an easy way to split the PR without having to use complicated git commands.
Assuming you want to split this PR into 2 PRs, you would keep some changes in this branch and move the others to the other branch. You would first create an identical branch to the one you are editing.
Then, on your original branch you would revert the changes that you only want to have in the other branch. And on that other branch you would revert the changes you want to keep only on the first branch. Then you would push your first branch again to update this PR, and you would push your second branch and create a new PR for it.

Wait, I just noticed that you made all your changes to your master branch. Here's a copy/paste of a comment I write from time to time:

> Hmm, I just noticed that your PR was based on commits that you pushed to your master branch.
While this is not a problem at all, many users prefer to use [feature-branches](https://github.com/haikuports/haikuports/wiki/DevelopmentModel#what-are-feature-branches).
This allows you to keep your master branch perfectly in sync with the project's master, and even easily refresh your master branch at any time, without having to touch your feature-branches.
That said, you are perfectly free to submit anything from your master branch. Using feature-branches is just a recommendation, and not using them does not have any impact.

Well, you can still work with this PR and your master branch. What you will need to do, after you create a second branch, will be to rebase that second branch against a git revision before your changes.
I can give you the git commands you will need to type, if you wish. (I just need to pull your branch and have a quick look at the hashes of your commits to tell you what you would need to type.)

The other solution, if you feel more comfortable, would be to close this PR and open 2 new PRs based on feature branches (but **not** using master this time :)
(Closing a PR to open a new one is not recommended but I guess this would be OK.)

fbrosson commented:
@korli, I used `git remote add` and `git fetch` to grab @khallebal's commits, and when I run `git log -p khallebal/master` I see some conflicts in `media-libs/libsndfile/libsndfile-1.0.28.recipe`. So splitting this PR won't be easy. Shall we ask @khallebal to close this PR?

@khallebal, if you close this PR, don't forget to make backups of the files you have changed.
Then please read about how feature branches work.

**AFTER** you have made backups of all the files you have modified, you can refresh you master branch with:
```
git branch -M master oldmaster
git checkout -b master 8faba33005c4eccb9135ad21e4b2d7773650484b
git pull upstream master
git push --force --set-upstream origin master
```
**WARNING** It will only be OK to run the command above if you have chosen to close this PR (to open a new one on a branch other than master.)

khallebal commented:
Yeah, it's probably the best solution, this PR too long already.
I'll try to find my way around the feature-branches, it'll take me some time to grasp the whole though, i'll close this PR once i figure it out.
